src/main/java/use_case/list/removeMedia/RemoveInteractor.java

```diff
@@ -0,0 +1,119 @@
+package use_case.list.removeMedia;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jetbrains.annotations.Nullable;
+
+import data_access.grade_api.UserRepository;
+import entity.AbstractMedia;
+import entity.MediaCollection;
+import entity.User;
+import use_case.list.ListOutputBoundary;
+import use_case.list.ListOutputData;
+import use_case.list.TableRowDataBuilder;
+
+/**
+ * The list display Interactor.
+ */
+public final class RemoveInteractor implements RemoveInputBoundary {
+    private final UserRepository userRepository;
+    private final ListOutputBoundary listPresenter;
+
+    public RemoveInteractor(UserRepository userRepository, ListOutputBoundary presenter) {
+        this.userRepository = userRepository;
+        this.listPresenter = presenter;
+    }
+
+    /**
+     * Entrypoint for calling the remove interactor to remove a media item from a collection.
+     * @param input The input data containing the collection name and the media name to be removed.
+     */
+    @Override
+    public void execute(RemoveMediaInput input) {
+        final User user = userRepository.get(userRepository.getCurrentUsername());
+        final List<MediaCollection<? extends AbstractMedia>> mediaCollections = user.getAllMediaCollections();
+        final MediaCollection<? extends AbstractMedia> collection = findCollectionWithName(
+                mediaCollections, input.getCollectionName());
+        final AbstractMedia mediaToRemove = findMediaByName(collection, input.getMediaName());
+
+        if (collection == null) {
+            listPresenter.prepareFailView(new ListOutputData(
+                    "Collection " + input.getCollectionName() + " not found.",
+                    getAvailableCollections(mediaCollections)
+            ));
+        }
+        else if (mediaToRemove == null) {
+            listPresenter.prepareFailView(new ListOutputData(
+                    "Media with the name " + input.getMediaName() + " not found in the collection.",
+                    getAvailableCollections(mediaCollections)
+            ));
+        }
+        else {
+            collection.removeMedia(mediaToRemove);
+            listPresenter.prepareSuccessView(new ListOutputData(
+                    collectionsToTableRowData(collection),
+                    collection.getName(),
+                    collection.getCollectionType(),
```

Comment:
The name of this argument in `ListOutputData`'s constructor and variable is too ambiguous and confusing, that's my bad. 
The `CollectionType` variable in `ListOutputData` is actually supposed to be the name of the type of the abstract media in the collection, and it's been renamed to be more accurate in my pull request (#32). 
So instead of `collection.getCollectionType()`, it should be `collection.getMediaType().getName()`.